### ISSUE TITLE
TypeScript errors

### ISSUE THREAD

**Issue by shashkovdanil** [kodadot/nft-gallery#8500]
I noticed that in many files there are TS errors. After `tsc --noEmit` I saw this

```
Found 112 errors in 50 files.
```

I think these errors are worth fixing, also update the types if they are written incorrectly. This will help improve the reliability and predictability of the code

@roiLeo @preschian wdyt?

**Comment by roiLeo** [comment]
>  can I setup codegen? 

I like zod

**Comment by roiLeo** [comment]
> Usually zod is used for validation, I have never seen it used to generate types for graphql. Can you share some example?

okay this is not related

> More info is here: https://the-guild.dev/graphql/codegen

How will you handle this since:
- schema is in external project
- schema can change
- schema can be different from chain

**Comment by roiLeo** [comment]
I know there is somewhere an issue about global (shared) schema between chains

may @vikiival give his opinion?

**Comment by vikiival** [comment]
> Do we have public link for schema? Or maybe we can fetch it?

```
https://squid.subsquid.io/stick/graphql
```

Replace `stick` with any of `snekk, speck, rubick, marck`

> I know there is somewhere an issue about global (shared) schema between chains

All is tracked here.

https://github.com/kodadot/loligo/issues

> global (shared) schema

Yup KodaDot currently uses `Universal Schema (v1)` which can guarantee that majority of fields are the same.

> I can do Proof of Concept

Let's try for AH* other chains 

> How will you handle this since:

This is true :) schema change is very hard to track, however if we can manage to cover USv1 it can solve some bugs

**Comment by vikiival** [comment]
Yes, I can update prod enpoint on the go, it helps us to fix stuff faster.

**Comment by roiLeo** [comment]
- link #4544
